datanommer.models/datanommer/models/__init__.py

```diff
@@ -128,7 +131,14 @@ def add(envelope):
     obj.msg = message['msg']
     obj.headers = headers
 
-    session.add(obj)
+    try:
+        session.add(obj)
+        session.flush()
```

Comment:
According to http://docs.sqlalchemy.org/en/latest/orm/session_api.html#sqlalchemy.orm.session.Session.flush flush() actually issues the INSERTs to the database, so if a row already exists in the database, it will raise an IntegrityError here, and it's handled appropriately.